packages/pubsub/src/lib/client.ts

```diff
@@ -1,19 +1,48 @@
 import type { Topic } from '@google-cloud/pubsub'
 import { PubSub } from '@google-cloud/pubsub'
 
-export const pubsub = new PubSub()
+export type ClientConfig = {
+  projectId: string
+}
+
+const localProjectId: string = 'local'
+const clients: Record<string, PubSub> = {}
+
+const init = (config?: ClientConfig) => {
```

Comment:
We could simplify this with a default `config`:

```ts
const init = ({ projectId }: ClientConfig = { projectId: localProjectId }) => {
  if (!clients[projectId]) {
    if (projectId === localProjectId) {
      // Create a default client when there is no config.
      clients[projectId] = new PubSub()
    } else {
      clients[projectId] = new PubSub({
        projectId,
      })
    }
  }
}
```

Reply:
Great idea!